File: core/src/de/project/ice/utils/DelegatingInputProcessor.java
package de.project.ice.utils;


import com.badlogic.gdx.InputProcessor;

/**
 * InputProcessor which just delegates all call to another InputProcessor
 * Usage: Subclass and selectively override Events your interested in
 */
public class DelegatingInputProcessor implements InputProcessor {
    private InputProcessor processor;

    public DelegatingInputProcessor(InputProcessor processor) {
        this.processor = processor;
    }

    @Override
    public boolean keyDown(int keycode) {
        return processor.keyDown(keycode);
    }

    @Override
    public boolean keyUp(int keycode) {
        return processor.keyUp(keycode);
    }

    @Override
    public boolean keyTyped(char character) {
        return processor.keyTyped(character);
    }

    @Override
    public boolean touchDown(int screenX, int screenY, int pointer, int button) {
        return processor.touchDown(screenX, screenY, pointer, button);
    }

    @Override
    public boolean touchUp(int screenX, int screenY, int pointer, int button) {
        return processor.touchUp(screenX, screenY, pointer, button);
    }

    @Override
    public boolean touchDragged(int screenX, int screenY, int pointer) {
        return processor.touchDragged(screenX, screenY, pointer);
    }

    @Override
    public boolean mouseMoved(int screenX, int screenY) {
        return processor.mouseMoved(screenX, screenY);
    }

    @Override
    public boolean scrolled(int amount) {
        return processor.scrolled(amount);
    }
}

File: core/src/de/project/ice/utils/DelegatingInputProcessor.kt
package de.project.ice.utils


import com.badlogic.gdx.InputProcessor

/**
 * InputProcessor which just delegates all call to another InputProcessor
 * Usage: Subclass and selectively override Events your interested in
 */
open class DelegatingInputProcessor(protected var processor: InputProcessor) : InputProcessor {
    override fun keyDown(keycode: Int): Boolean = processor.keyDown(keycode)
    override fun keyUp(keycode: Int): Boolean = processor.keyUp(keycode)
    override fun keyTyped(character: Char): Boolean = processor.keyTyped(character)
    override fun touchDown(screenX: Int, screenY: Int, pointer: Int, button: Int): Boolean = processor.touchDown(screenX, screenY, pointer, button)
    override fun touchUp(screenX: Int, screenY: Int, pointer: Int, button: Int): Boolean = processor.touchUp(screenX, screenY, pointer, button)
    override fun touchDragged(screenX: Int, screenY: Int, pointer: Int): Boolean = processor.touchDragged(screenX, screenY, pointer)
    override fun mouseMoved(screenX: Int, screenY: Int): Boolean = processor.mouseMoved(screenX, screenY)
    override fun scrolled(amount: Int): Boolean = processor.scrolled(amount)
}
